hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogStateManagerImpl.java

```diff
@@ -213,6 +214,7 @@ public void increaseRetryCountOfTransactionInDB(
   public int resetRetryCountOfTransactionInDB(ArrayList<Long> txIDs)
       throws IOException {
     for (long txId: txIDs) {
+      Objects.requireNonNull(txId, "transactionId cannot be null");
```

Comment:
This `requireNonNull` check is unnecessary.

`txIDs` may be `null`, or it may contain a `null` element.  In both cases `for (long txId: txIDs)` would fail before reaching `requireNonNull`.

Reply:
You are right.  I should wait for Lokeshj to give a clarification on the possible NPE. From my view, the lock should have prevented resetting the nonexisting txIds.

Reply:
I meant handling NPE for this code.
`deletedTable.get(txId);`
The table may not have entry for a particular transactionID.

Reply:
You are right, I find the case in the other place and applied it here now.